src/d/actor/d_a_deku_item.cpp

```diff
@@ -4,106 +4,277 @@
+    JUT_ASSERT(0xF4, modelData != 0);
+
+    mpModel = mDoExt_J3DModel__create(modelData, 0, 0x11020203U);
+
+    if (!mpModel) {
+        return FALSE;
+    }
+
+    J3DAnmTransform* pbck = (J3DAnmTransform*)dComIfG_getObjectRes(m_arcname, DEKU_BCK_VLFDK);
+    JUT_ASSERT(0x103, pbck != 0);
+
+    if (!mBck1.init(modelData, pbck, TRUE, J3DFrameCtrl::LOOP_REPEAT_e, 1.0f, 0, -1, false)) {
+        return FALSE;
+    }
+
+    pbck = (J3DAnmTransform*)dComIfG_getObjectRes(m_arcname, DEKU_BCK_VLFDM);
+    JUT_ASSERT(0x110, pbck != 0);
+
+    if (!mBck2.init(modelData, pbck, TRUE, J3DFrameCtrl::LOOP_ONCE_e, 1.0f, 0, -1, false)) {
+        return FALSE;
+    }
+
+    return TRUE;
 }
 
 /* 000002C8-000003A8       .text CreateInit__12daDekuItem_cFv */
 void daDekuItem_c::CreateInit() {
-    /* Nonmatching */
+    fopAcM_SetMtx(this, mpModel->getBaseTRMtx());
+
+    fopAcM_setCullSizeBox(this, -50.0f, 0.0f, -50.0f, 50.0f, 100.0f, 50.0f);
+    mStts.Init(0xFF, 0xFF, this);
+
+    mCyl.Set(l_cyl_src);
+    mCyl.SetStts(&mStts);
+    mAcchCir.SetWall(50.0f, 50.0f);
+    mAcch.Set(fopAcM_GetPosition_p(this), fopAcM_GetOldPosition_p(this), this, 1, &mAcchCir, fopAcM_GetSpeed_p(this));
+
+    set_mtx();
+
+    mMode = Mode_WAIT_e;
+    fopAcM_SetGravity(this, -3.0f);
 }
 
 /* 000003A8-00000598       .text _create__12daDekuItem_cFv */
 s32 daDekuItem_c::_create() {
-    /* Nonmatching */
+    fopAcM_SetupActor(this, daDekuItem_c);
+
+    mItemBitNo = daDekuItem_prm::getItemBitNo(this);
+
+    if (fopAcM_isItem(this, mItemBitNo)) {
+        return cPhs_ERROR_e;
+    }
+
+    if (!dComIfGs_isEventBit(0x1801U)) {
+        return cPhs_ERROR_e;
+    }
+
+    s32 phase_state = dComIfG_resLoad(&mPhs, m_arcname);
+
+    if (phase_state == cPhs_COMPLEATE_e) {
+        if (!fopAcM_entrySolidHeap(this, CheckCreateHeap, 0x5000)) {
+            return cPhs_ERROR_e;
+        }
+
+        CreateInit();
+    }
+
+    return phase_state;
 }
 
 /* 00000948-000009C8       .text set_mtx__12daDekuItem_cFv */
 void daDekuItem_c::set_mtx() {
-    /* Nonmatching */
+    mpModel->setBaseScale(scale);
+    mDoMtx_stack_c::transS(current.pos.x, current.pos.y, current.pos.z);
+    mDoMtx_stack_c::YrotM(current.angle.y);
+
+    mpModel->setBaseTRMtx(mDoMtx_stack_c::get());
 }
 
 /* 000009C8-00000A74       .text _execute__12daDekuItem_cFv */
 BOOL daDekuItem_c::_execute() {
-    /* Nonmatching */
+    eyePos = current.pos;
+    fopAcM_posMoveF(this, mStts.GetCCMoveP());
+    mAcch.CrrPos(*dComIfG_Bgsp());
+
+    checkOrder();
+    mode_proc_call();
+    eventOrder();
+
+    set_mtx();
+    mBck1.setFrame(mBck1.getEndFrame());
+
+    return TRUE;
 }
 
 /* 00000A74-00000B30       .text mode_proc_call__12daDekuItem_cFv */
 void daDekuItem_c::mode_proc_call() {
-    /* Nonmatching */
+    static ModeFunc mode_proc[] = {
+        &mode_wait,
+        &mode_getdemo,
+        &mode_getdemo_init,
+        &mode_getdemo_wait,
+    };
```

Comment:
```suggestion
    static ModeFunc mode_proc[] = {
        &mode_wait,
        &mode_getdemo_init,
        &mode_getdemo_wait,
        &mode_getdemo,
    };
```

noticed the reason it's not matching. these PTMFs are out of order which makes the relocations not match.
![image](https://github.com/zeldaret/tww/assets/19174152/836fdb94-967c-46cf-bc6b-34ef2c3c722c)

after changing this, you also need to fix the Mode_e enum and everything using it, since the names are in the same order as this array

Reply:
Thanks so much. I have no idea how I missed that.